modules/medianetBidAdapter.js

```diff
@@ -1,8 +1,68 @@
 import { registerBidder } from 'src/adapters/bidderFactory';
 import * as utils from 'src/utils';
+import events from 'src/events';
+import CONSTANTS from 'src/constants.json';
+import { config } from 'src/config';
+import * as url from 'src/url';
 
 const BIDDER_CODE = 'medianet';
 const BID_URL = 'https://prebid.media.net/rtb/prebid';
+const EVENT_PIXEL_URL = 'lg1.media.net/log';
+const { BID_TIMEOUT } = CONSTANTS.EVENTS;
+const TIMEOUT_EVENT_NAME = 'client_timeout';
+
+let bidParams = {};
+
+events.on(BID_TIMEOUT, function (timedOutBidders) {
```

Comment:
Listening to and logging event data isn't allowed in a bidder adapter, but is in an [analytics adapter](http://prebid.org/dev-docs/integrate-with-the-prebid-analytics-api.html).

Reply:
Hey @matthewlane ,

Shouldn't the bidder timeout be allowed in the main adapter, the analytics adapter could listen to a lot more events based on the environment. The publisher might not always pick the analytics adapter, as it generally leads to more px calls on the browser.

However the client timeout is an important event that the bidder **must** listen to to optimize, tweak things at its end. This optimization could directly lead to better yields to the publisher.

Your views?

Thanks,
Ruturaj

Reply:
Makes sense. The `BID_TIMEOUT` event still shouldn't be listened to in bidder adapters as it may contain data from other bidders, even though you are filtering them out below, but I've create an issue to define and implement a way for bidders to subscribe to their own timed out event data: https://github.com/prebid/Prebid.js/issues/2254. Feel free to comment on that or use it for tracking and whenever a PR is merged you'll be able to use it here